### PR TITLE
Fix/proposal to store code out of gas error

### PR DESCRIPTION
**Pull request by poomthiti**
## Describe your changes
1. fix out of gas error by gzipping wasm byte code before submitting tx
2. get the correct precision for staking assets from chain record

### CODE REVIEW

File: src/lib/app-provider/hooks/useChainRecordAsset.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        (asset) => asset.base === denom
+      );
+      return assetInfo
+        ? (snakeToCamel({
+            ...assetInfo,
+            precision:
+              assetInfo?.denom_units.find(
+                (unit) =>
+                  unit.denom ===
+                  (denom[0] === "u" ? denom.slice(1) : "u".concat(denom))
```
Comment (bkioshn) [inline src/lib/app-provider/hooks/useChainRecordAsset.ts:29]: I think "u".concat(denom) is quite strange. maybe just return only denom instead or add todo for revisiting

Comment (songwongtp) [reply src/lib/app-provider/hooks/useChainRecordAsset.ts:29]: the logic is a bit weird here -- if "denom" starts with "u", it is sliced out. But if "denom" doesn't start with "u", "u" is appended to the front of the denom

Comment (poomthiti) [reply src/lib/app-provider/hooks/useChainRecordAsset.ts:29]: in chain record the exponent field is inverse of precision 
e.g. (Precision) uToken = 6 token = 0,  (Exponent) uToken = 0 token = 6

Comment (bkioshn) [reply src/lib/app-provider/hooks/useChainRecordAsset.ts:29]: Understand what you mean, just worry about the case where there is no uToken and token or other edge case

**Pull request merged**